application/apps/indexer/addons/someip-tools/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+use std::{
+    fmt::{self, Display},
+    net::Ipv4Addr,
+};
+
+use nom::{
+    combinator::map,
+    number::streaming::{be_u16, be_u32, be_u8},
+    sequence::tuple,
+    IResult,
+};
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Parse error: {0}")]
+    Parse(String),
+    #[error("Incomplete, not enough data for a message")]
+    Incomplete,
+}
+
+impl nom::error::ParseError<&[u8]> for Error {
+    fn from_error_kind(input: &[u8], kind: nom::error::ErrorKind) -> Self {
+        Error::Parse(format!(
+            "Nom error: {:?} ({} bytes left)",
+            kind,
+            input.len()
+        ))
+    }
+
+    fn append(_: &[u8], _: nom::error::ErrorKind, other: Self) -> Self {
+        other
+    }
+}
+
+/// Parses a DLT Network-Trace prefix for a SOME/IP message from the given input.
+///
+/// A valid prefix will consist of:
+/// - 4 bytes as IPv4 address
+/// - 2 bytes as udp/tcp port
+/// - 1 byte as protocol type (0 = local, 1 = tcp, 2 = udp)
+/// - 1 byte as message direction (0 = incoming, 1 = outgoing)
+/// - 1, 2 or 4 bytes as instance-id
+pub fn parse_prefix(input: &[u8]) -> IResult<&[u8], std::string::String, Error> {
```

Comment:
best not to leak library dependent types in public APIs.
easily replaced by a general `Result`
```rust
pub type IResult<I, O, E = Error<I>> = Result<(I, O), Err<E>>;
```

Reply:
done.